src/measure_theory/measure/haar.lean

```diff
@@ -644,6 +644,55 @@ begin
   exact regular.smul ctop,
 end
 
+/-- **Steinhaus Theorem** In any locally compact group `G` with a haar measure `μ`, for any
+  measurable set `E` of positive measure, the set `E/E` contains a neighborhood of `1`. -/
+@[to_additive "**Steinhaus Theorem** In any locally compact group `G` with a haar measure `μ`,
+  for any measurable set `E` of positive measure, the set `E - E` contains a neighborhood of `0`."]
+theorem steinhaus_theorem_mul (μ : measure G) [is_haar_measure μ] [locally_compact_space G]
+  (E : set G) (hE : measurable_set E) (hEpos : 0 < μ E) :
+  ∃ (U : set G) , U ∈ nhds (1 : G) ∧ U ⊆ E / E :=
+begin
+  rcases (exists_subset_measure_lt_top hE hEpos) with ⟨L, hL, hLE, hLpos, hLtop⟩,
```

Comment:
Since something nontrivial is going on in this proof, it is worthy to make it more readable. Here are two ways that you can use for this:
* At the beginning of the proof, you can give an informal sketch of the proof in a comment of a few lines, using the same notations as in the formal proof to follow. That helps a lot understanding what is going on!
* Instead of `rcases ... with ...`, you can use `obtain ... : ..., from ...`, specifying the type of the objects you are defining. For instance, the first three lines of your proof could be
```lean
  obtain ⟨L, hL, hLE, hLpos, hLtop⟩ : ∃ (L : set G), measurable_set L ∧ L ⊆ E ∧ 0 < μ L ∧ μ L < ⊤,
    from exists_subset_measure_lt_top hE hEpos,
  obtain ⟨K, hKL, hK, hKpos⟩ : ∃ (K : set G) (H : K ⊆ L), is_compact K ∧ 0 < μ K,
    from measurable_set.exists_lt_is_compact_of_ne_top hL (ne_of_lt hLtop) hLpos,
```

Reply:
Thanks for this comment! Your suggestion indeed makes the proofs much more readable. I have taken your suggestion up in the new commit!